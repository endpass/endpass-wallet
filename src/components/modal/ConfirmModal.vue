<template>
  <v-modal
    class="is-dark"
    @close="handleClose"
  >
    <template slot="header">Are you sure?</template>

    <slot />

    <div
      slot="footer"
      class="buttons"
    >
      <v-button
        class-name="is-primary is-medium"
        data-test="submit-logout"
        @click="handleConfirm"
      >
        Confirm
      </v-button>
      <v-button
        class-name="is-medium"
        data-test="submit-logout"
        @click="handleClose"
      >
        Cancel
      </v-button>
    </div>
  </v-modal>
</template>

<script>
import VModal from '@/components/ui/VModal';
import VButton from '@/components/ui/form/VButton';

export default {
  name: 'ConfirmModal',
  methods: {
    handleConfirm() {
      this.$emit('confirm');
    },
    handleClose() {
      this.$emit('close');
    },
  },
  components: {
    VButton,
    VModal,
  },
};
</script>
