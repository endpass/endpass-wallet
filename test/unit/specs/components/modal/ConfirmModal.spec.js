import { shallowMount, createLocalVue } from '@vue/test-utils';
import UIComponents from '@endpass/ui';

import ConfirmModal from '@/components/modal/ConfirmModal';

const localVue = createLocalVue();

localVue.use(UIComponents);

describe('ConfirmModal', () => {
  let wrapper;

  describe('render', () => {
    beforeEach(() => {
      wrapper = shallowMount(ConfirmModal, {
        localVue,
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('ConfirmModal');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    beforeEach(() => {
      wrapper = shallowMount(ConfirmModal, {
        localVue,
      });
    });

    it('should emit confirm event', () => {
      wrapper.vm.handleConfirm();

      expect(wrapper.emitted().confirm).toHaveLength(1);
    });

    it('should emit close event', () => {
      wrapper.vm.handleClose();

      expect(wrapper.emitted().close).toHaveLength(1);
    });
  });
});
