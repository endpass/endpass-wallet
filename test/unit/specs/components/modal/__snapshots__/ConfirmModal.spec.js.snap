// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal render should render component 1`] = `
<v-modal
  class="is-dark"
  v-on:close="bound handleClose"
>
  <template
    slot="header"
  >
    Are you sure?
  </template>
    
  <div
    class="buttons"
    slot="footer"
  >
    <v-button
      class-name="is-primary is-medium"
      data-test="submit-logout"
      v-on:click="bound handleConfirm"
    >
      
      Confirm
    
    </v-button>
     
    <v-button
      class-name="is-medium"
      data-test="submit-logout"
      v-on:click="bound handleClose"
    >
      
      Cancel
    
    </v-button>
  </div>
</v-modal>
`;
